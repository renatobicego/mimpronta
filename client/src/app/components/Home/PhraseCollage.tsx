import Image from "next/image";
import React from "react";

const PhraseCollage = () => {
  return (
    <div className="relative w-full h-screen lg:h-auto lg:min-h-screen flex flex-col items-center">
      <Image
        src={"/papeles/papelMarron1.png"}
        alt="papel marron 1"
        quality={75}
        width={250}
        height={700}
        className="absolute -left-2 top-[10%] lg:left-0 
                        rotate-180 -scale-x-100 h-[35%] xsm:h-[40%] w-auto lg:h-auto"
      />
      <h1
        role="heading"
        id="phraseCollageMundo"
        className="phrase-size leading-phrase text-center font-semibold 
            size-section sm:w-[80%] lg:w-[65%] 2xl:w-[60%] relative z-10
            mt-48 md:mt-[30vh] lg:mt-[22vh] lg:mb-4 3xl:mt-56"
      >
        El mundo atraviesa momentos desafiantes, y con tu
        <span className="font-extrabold"> impronta </span> puedes transformarte
        en{" "}
        <span className="font-extrabold text-rosaOscuro">
          {" "}
          agente de cambio
        </span>
      </h1>
      <Image
        id="collageMundo"
        role="img"
        quality={30}
        src={"/collages/mundo.png"}
        alt="Collage mundo"
        priority
        width={1655}
        height={1055}
        className="scale-[1.35] top-[55vh] md:mt-[15vh] lg:mt-6 w-auto overflow-auto md:scale-[1.15] lg:scale-100 lg:w-[65%]
                        absolute md:top-0 md:relative z-40"
      />
    </div>
  );
};

export default PhraseCollage;
