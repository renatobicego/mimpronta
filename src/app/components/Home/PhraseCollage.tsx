import Image from 'next/image'
import React from 'react'

const PhraseCollage = () => {
  return (
    <>
        <Image 
            src={'/papeles/papelMarron1.png'} 
            alt='papel marron 1'
            width={250}
            height={700} 
            className='absolute -left-2 lg:left-0 top-[5%] 
                        rotate-180 -scale-x-100 h-[35%] xsm:h-[40%] w-auto lg:h-auto
                        lg:rotate-0 lg:scale-x-100'
            />
        <h3 role='heading' className='phrase-size leading-phrase text-center font-semibold 
            size-section lg:w-[65%] 2xl:w-[55%] 
            mt-48 xsm:mt-52 md:mt-[30vh] lg:mt-40 lg:mb-4'>
            El mundo atraviesa momentos 
            desafiantes, y cada persona con
            su <span className='font-extrabold'> impronta </span> puede transformarse
            en <span className='font-extrabold'> agente de cambio</span>
        </h3>
        <Image 
            role='img'
            src={'/collages/mundo.png'}
            alt='Collage mundo'
            priority
            width={1000}
            height={800}
            className='scale-[1.35] mt-[13vh] xsm:mt-[27vh] lg:mt-0 lg:w-auto overflow-auto lg:scale-100
                        relative z-40'
            />
    </>
  )
}

export default PhraseCollage