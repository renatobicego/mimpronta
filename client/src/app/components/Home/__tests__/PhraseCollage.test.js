import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'
import PhraseCollage from '../PhraseCollage.tsx'

describe('About Us', () => {
    it('renders unchanged', () => {
        const {container} = render(<PhraseCollage />)
        expect(container).toMatchSnapshot()
    })

    it('renders content', async() => {
        render(<PhraseCollage />)
        const title = await screen.findByRole('heading')
        const collage = await screen.findAllByRole('img')

        expect(title).toBeInTheDocument()
        expect(collage).toHaveLength(3)
    })

});