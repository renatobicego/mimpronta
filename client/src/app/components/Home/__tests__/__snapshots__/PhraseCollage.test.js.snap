// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Us renders unchanged 1`] = `
<div>
  <img
    alt="papel marron 1"
    class="absolute -left-2 md:left-0 top-[5%] lg:top-[8%]    rotate-180 -scale-x-100 h-[60%] w-auto lg:h-auto   lg:rotate-0 lg:scale-x-100"
    data-nimg="1"
    decoding="async"
    height="700"
    loading="lazy"
    src="/_next/image?url=%2Fpapeles%2FpapelMarron1.png&w=640&q=75"
    srcset="/_next/image?url=%2Fpapeles%2FpapelMarron1.png&w=256&q=75 1x, /_next/image?url=%2Fpapeles%2FpapelMarron1.png&w=640&q=75 2x"
    style="color: transparent;"
    width="250"
  />
  <img
    alt="papel marron 2"
    class="absolute left-[88dvw] lg:left-[90vw] top-[10%] lg:top-[120%]    rotate-[185deg] -scale-x-100 h-[60%] w-auto lg:h-auto   lg:rotate-0 lg:scale-x-100"
    data-nimg="1"
    decoding="async"
    height="1200"
    loading="lazy"
    src="/_next/image?url=%2Fpapeles%2FpapelMarron4.png&w=1080&q=75"
    srcset="/_next/image?url=%2Fpapeles%2FpapelMarron4.png&w=640&q=75 1x, /_next/image?url=%2Fpapeles%2FpapelMarron4.png&w=1080&q=75 2x"
    style="color: transparent;"
    width="500"
  />
  <h3
    class="phrase-size leading-phrase text-center font-semibold    size-section lg:w-[75%] xl:w-[65%] 2xl:w-[55%]    mt-16 xsm:mt-20 md:mt-28 lg:mt-16 lg:mb-4"
    role="heading"
  >
    El mundo atraviesa momentos 
    <br
      class="hidden xsm:block lg:hidden"
    />
    desafiantes, y cada persona con 
    <br
      class="hidden xsm:block lg:hidden"
    />
    su 
    <span
      class="font-extrabold"
    >
       impronta 
    </span>
     puede transformarse 
    <br
      class="hidden xsm:block lg:hidden"
    />
    en 
    <span
      class="font-extrabold"
    >
       agente de cambio
    </span>
  </h3>
  <img
    alt="Collage mundo"
    class="-z-10 scale-[1.35] mt-[13dvh] xsm:mt-[27dvh] lg:mt-[10dvh] lg:w-auto overflow-auto lg:scale-[1.15]"
    data-nimg="1"
    decoding="async"
    fetchpriority="high"
    height="800"
    role="img"
    src="/_next/image?url=%2Fcollages%2Fmundo.png&w=2048&q=75"
    srcset="/_next/image?url=%2Fcollages%2Fmundo.png&w=1080&q=75 1x, /_next/image?url=%2Fcollages%2Fmundo.png&w=2048&q=75 2x"
    style="color: transparent;"
    width="1000"
  />
</div>
`;
